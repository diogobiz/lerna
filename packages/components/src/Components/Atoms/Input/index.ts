export { default as SANInput, ISANInputProps } from './Input'
