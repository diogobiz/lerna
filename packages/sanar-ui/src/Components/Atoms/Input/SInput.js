import React, { memo, forwardRef } from 'react'
import PropTypes from 'prop-types'

import styled, { css } from 'styled-components'
import { theme, ifProp, switchProp } from 'styled-tools'

const Input = styled.input`
    border: 1px solid ${theme('colors.grey.2')};
    box-shadow: 0 1px 2px ${theme('colors.grey.1')};
    border-radius: ${theme('radii.base')}px;
    outline: none;
    color: ${theme('colors.grey-solid.7')};
    background-color: transparent;
    font-weight: ${theme('fontWeights.medium')};
    font-size: ${theme('fontSizes.md')}px;
    transition: all 0.5s cubic-bezier(0.075, 0.82, 0.165, 1);
    width: 100%;

    ${switchProp('size', {
        large: css`
            padding: ${theme('space.xs')}px;
        `,
        medium: css`
            padding: ${theme('space.1')}px;
        `,
        small: css`
            padding: ${theme('space.0')}px;
        `
    })}

    ${ifProp(
        'round',
        css`
            border-radius: 50vh;
        `,
        css`
            border-radius: ${theme('radii.base')}px;
        `
    )}

    &:disabled {
        background-color: ${theme('colors.grey-solid.2')};
        opacity: 0.5;
    }

    &:hover:not(:disabled) {
        border-color: ${theme('colors.grey.5')};
    }

    &:focus {
        border-color: ${theme('colors.blue.1')};
        caret-color: ${theme('colors.blue.1')};
    }

    ::placeholder {
        color: ${theme('colors.grey-solid.5')};
    }

    ::autofill {
        background-color: transparent;
    }
`

const Wrapper = styled.div`
    position: relative;
    display: inline-flex;
    align-items: center;
    width: 100%;

    & .es-eva-icon {
        position: absolute;
        color: ${theme('colors.grey.6')};
    }

    ${ifProp(
        'iconLeft',
        css`
            & .es-eva-icon:first-child {
                left: ${theme('space.sm')}px;
            }
        `
    )}

    ${ifProp(
        'disabled',
        css`
            & .es-eva-icon {
                color: ${theme('colors.grey-solid.3')};
            }
        `
    )}

    ${ifProp(
        'iconRight',
        css`
            & .es-eva-icon:last-child {
                right: ${theme('space.sm')}px;
            }
        `
    )}

    ${Input} {
        ${ifProp(
            'iconLeft',
            css`
                padding-left: 36px;
            `
        )}
        ${ifProp(
            'iconRight',
            css`
                padding-right: 36px;
            `
        )}
    }
`

export const SInput = memo(
    forwardRef(
        ({ placeholder, iconRight, iconLeft, disabled, ...props }, ref) => {
            const customPlaceholder = props.required
                ? `${placeholder} *`
                : placeholder

            return (
                <Wrapper {...{ iconRight, iconLeft, disabled }}>
                    {iconLeft && iconLeft}
                    <Input
                        ref={ref}
                        {...{ ...props, disabled }}
                        placeholder={customPlaceholder}
                    />
                    {iconRight && iconRight}
                </Wrapper>
            )
        }
    )
)

SInput.propTypes = {
    disabled: PropTypes.bool,
    value: PropTypes.string,
    placeholder: PropTypes.string,
    onChange: PropTypes.func,
    size: PropTypes.oneOf(['large', 'medium', 'small']),
    round: PropTypes.bool
}

SInput.defaultProps = {
    autoComplete: 'off',
    size: 'large'
}
