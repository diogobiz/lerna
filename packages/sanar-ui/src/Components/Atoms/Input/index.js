export { default } from './Input'
export { default as ESInputSearch } from './InputSearch'
export { default as ESTextArea } from './TextArea'
