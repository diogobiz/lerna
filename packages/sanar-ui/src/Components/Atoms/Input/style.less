& .es-text-area {
    box-shadow: 0 1px 2px @grey-3;
}

.es-input {
    &__dark,
    &-search__dark {
        input {
            color: @white-5;
            border-color: @white-3;
            background-color: @white-1;
        }
        & .anticon {
            color: @white-7;
        }
    }
}
