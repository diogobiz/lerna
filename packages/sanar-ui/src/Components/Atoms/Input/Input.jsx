import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import Input from 'antd/lib/input'
import classNames from 'classnames'

import warning from '../../../Util/Warning'

const ESInput = ({ className, dark, component, ...props }) => {
    const classes = classNames(
        'es-input',
        { 'es-input__dark': dark },
        className
    )

    const Comp = component ? component : Input

    useEffect(() => {
        warning(
            false,
            'ESInput',
            'ESInput component is deprecated. Please use SInput component.'
        )
    }, [])

    return <Comp className={classes} {...props} />
}

ESInput.propTypes = Object.assign(
    { ...Input['propTypes'] },
    {
        className: PropTypes.string,
        dark: PropTypes.bool
    }
)

ESInput.defaultProps = Input['defaultProps']

ESInput.Password = Input.Password

export default ESInput
