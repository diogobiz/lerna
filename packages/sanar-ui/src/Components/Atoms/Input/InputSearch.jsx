import React from 'react'
import PropTypes from 'prop-types'
import Input from 'antd/lib/input'
import classNames from 'classnames'

const Search = Input.Search

const ESInputSearch = ({ className, dark, ...props }) => {
    const classes = classNames(
        'es-input-search',
        { 'es-input-search__dark': dark },
        className
    )

    return <Search className={classes} {...props} />
}

ESInputSearch.propTypes = Object.assign(
    { ...Search['propTypes'] },
    {
        className: PropTypes.string,
        onSearch: PropTypes.func,
        dark: PropTypes.bool
    }
)

ESInputSearch.defaultProps = Search['defaultProps']

export default ESInputSearch
