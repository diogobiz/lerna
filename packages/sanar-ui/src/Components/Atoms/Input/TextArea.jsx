import React from 'react'
import PropTypes from 'prop-types'
import Input from 'antd/lib/input'
import classNames from 'classnames'

const TextArea = Input.TextArea

const ESTextArea = ({ className, ...props }) => {
    const classes = classNames('es-text-area', className)

    return <TextArea className={classes} {...props} />
}

ESTextArea.propTypes = Object.assign(
    { ...TextArea['propTypes'] },
    {
        className: PropTypes.string,
        autosize: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
        defaultValue: PropTypes.string,
        value: PropTypes.string,
        onPressEnter: PropTypes.func
    }
)

ESTextArea.defaultProps = TextArea['defaultProps']

export default ESTextArea
