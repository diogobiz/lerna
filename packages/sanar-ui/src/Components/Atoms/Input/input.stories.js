import React from 'react'
import { storiesOf } from '@storybook/react'
import { select, text, boolean } from '@storybook/addon-knobs'

import ESInput, { ESInputSearch, ESTextArea } from './'
import { SInput } from './SInput'
import ESEvaIcon from '../EvaIcon'
import ESIcon from '../Icon'

const sizeOptions = {
    Default: 'default',
    Large: 'large',
    Small: 'small'
}

const sizeSOptions = {
    Large: 'large',
    Medium: 'medium',
    Small: 'small'
}

storiesOf('Atoms.Input', module)
    .add('Simple', () => (
        <ESInput
            disabled={boolean('Disabled', false)}
            placeholder={text('Placeholder', 'E-mail')}
            prefix={<ESIcon type='mail' />}
            size={select('Size', sizeOptions, 'default')}
            addonAfter={text('Addon after', 'addonAfter')}
            addonBefore={text('Addon before', 'addonBefore')}
            allowClear={boolean('Allow clear', false)}
        />
    ))
    .add('Password', () => (
        <ESInput
            placeholder={text('Placeholder', 'Senha')}
            component={ESInput.Password}
        />
    ))
    .add('Search', () => (
        <ESInputSearch
            placeholder={text('Placeholder', 'Busque seu conteÃºdo')}
            enterButton={text('Enter button', '')}
        />
    ))
    .add('Text Area', () => <ESTextArea />)
    .add('Styled', () => (
        <SInput
            disabled={boolean('Disabled', false)}
            placeholder={text('Placeholder', 'Placeholder')}
            iconLeft={<ESEvaIcon name='alert-circle-outline' />}
            iconRight={<ESEvaIcon name='alert-circle-outline' />}
            size={select('Size', sizeSOptions, 'large')}
            required={boolean('Required', false)}
            round={boolean('Round', false)}
        />
    ))
