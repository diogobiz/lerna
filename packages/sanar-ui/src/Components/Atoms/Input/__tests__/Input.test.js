import React from "react";
import TestRenderer from "react-test-renderer";

import Input from "antd/lib/input";
import ESInput, { ESInputSearch } from "../";
import ESIcon from '../../Icon'

it("renders correctly", () => {
	const component = <ESInput placeholder="E-mail" prefix={<ESIcon type="mail" />} />;

	const tree = TestRenderer.create(component).toJSON();
	expect(tree).toMatchSnapshot();
});

it("renders correctly", () => {
	const component = <ESInput placeholder="Senha" component={Input.Password} />;

	const tree = TestRenderer.create(component).toJSON();
	expect(tree).toMatchSnapshot();
});

it("renders correctly", () => {
	const component =  <ESInputSearch placeholder="Busque seu conteÃºdo" />;

	const tree = TestRenderer.create(component).toJSON();
	expect(tree).toMatchSnapshot();
});
