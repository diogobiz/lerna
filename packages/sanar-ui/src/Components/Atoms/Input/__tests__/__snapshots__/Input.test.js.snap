// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="es-input ant-input-affix-wrapper"
>
  <span
    className="ant-input-prefix"
  >
    <i
      aria-label="icon: mail"
      className="anticon anticon-mail es-icon"
      style={
        Object {
          "fontSize": undefined,
        }
      }
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="mail"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
        />
      </svg>
    </i>
  </span>
  <input
    className="ant-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="E-mail"
    style={null}
    type="text"
    value=""
  />
</span>
`;

exports[`renders correctly 2`] = `
<span
  className="ant-input-password es-input ant-input-affix-wrapper"
>
  <input
    action="click"
    className="ant-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Senha"
    style={null}
    type="password"
    value=""
  />
  <span
    className="ant-input-suffix"
  >
    <i
      aria-label="icon: eye-invisible"
      className="anticon anticon-eye-invisible ant-input-password-icon"
      onClick={[Function]}
      onMouseDown={[Function]}
      tabIndex={-1}
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="eye-invisible"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
        />
        <path
          d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
        />
      </svg>
    </i>
  </span>
</span>
`;

exports[`renders correctly 3`] = `
<span
  className="ant-input-search es-input-search ant-input-affix-wrapper"
>
  <input
    className="ant-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Busque seu conteÃºdo"
    style={null}
    type="text"
    value=""
  />
  <span
    className="ant-input-suffix"
  >
    <i
      aria-label="icon: search"
      className="anticon anticon-search ant-input-search-icon"
      onClick={[Function]}
      tabIndex={-1}
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="search"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
        />
      </svg>
    </i>
  </span>
</span>
`;
